### PR TITLE
Simple Example of a Homomorphic Program

### PR DESCRIPTION
**Pull request by rbharath**
This PR adds a simple example of a homomorphic program that performs computations on encrypted random data. There are a number of parameters that are used in a homomorphic encryption scheme, which the simple program attempts to explain in its comments.

Note that this example doesn't do computation on user-input data. Homomorphic schemes require that their input is structured a particular way which makes it tricky to input raw binary data. A subsequent PR will demonstrate how to compute on arbitrary input binary data.

### CODE REVIEW

File: src/PrintEncrypted.cpp [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+namespace std {} using namespace std;
+
+int main(int argc, char *argv[])
+{
+  ArgMapping amap;
+
+  // Technical parameter. Use if you want to explicitly set generator elements
+  // for the plaintext arrays.
+  Vec<long> gens;
```
Comment (robrobbins) [inline src/PrintEncrypted.cpp]: in the future just use the long, more descriptive variable name. 

`Vec<long> generators` etc...

yeah, its not as succinct but makes groking it easier for the other guy

( also feel free to bust me for it too :) )

File: src/PrintEncrypted.cpp [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+  amap.arg("gens", gens, "use specified vector of generators", NULL);
+  amap.note("e.g., gens='[562 1871 751]'");
+
+  // Technical parameter. The order of each generator specified in ords.
+  Vec<long> ords;
```
Comment (robrobbins) [inline src/PrintEncrypted.cpp]: same. what's an ord?

File: src/PrintEncrypted.cpp [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+  amap.arg("ords", ords, "use specified vector of orders", NULL);
+  amap.note("e.g., ords='[4 2 -4]', negative means 'bad'");
+
+  // Random seed used in scheme.
+  long seed=0;
+  amap.arg("seed", seed, "PRG seed");
+
+  // The number of rounds of encrypted computation. If R > 1, then we need to
+  // "bootstrap" between rounds which adds a heavy computational overhead.
+  long R=1;
```
Comment (robrobbins) [inline src/PrintEncrypted.cpp]: now i'm really gunna complain...

File: src/PrintEncrypted.cpp (hunk 1 above)
Comment (rbharath) [reply src/PrintEncrypted.cpp]: Good suggestion. I copied over name defaults from their test files, and swapping over to readable names seems better practice.

File: src/PrintEncrypted.cpp (hunk 2 above)
Comment (rbharath) [reply src/PrintEncrypted.cpp]: Good suggestion, will add in a comment and change the name.

File: src/PrintEncrypted.cpp (hunk 3 above)
Comment (rbharath) [reply src/PrintEncrypted.cpp]: :-). Will try to add in descriptive names for these one-char variable names instead.

**Comment by rbharath** [comment]
@robrobbins Went through and used more descriptive variable names. PTAL

**Comment by rbharath** [comment]
Comments on refactoring to be done added here: https://github.com/computablelabs/HElib/issues/7

**Comment by rbharath** [comment]
Based on offline discussion with @robrobbins, going to go ahead and merge this PR in. Will handle refactoring efforts in future PRs.

**Pull request merged**